SwiftGit2/CommitIterator.swift

```diff
@@ -0,0 +1,67 @@
+//
+// Created by Arnon Keereena on 4/28/17.
+// Copyright (c) 2017 GitHub, Inc. All rights reserved.
+//
+
+import Result
+import libgit2
+
+public class CommitIterator: IteratorProtocol {
+    public typealias Element = Result<Commit, NSError>
+    var repo: Repository
+    var branch: Branch
+    var revisionWalker: OpaquePointer? = nil
+    var oid: git_oid
+    private var unsafeCommit: OpaquePointer? = nil
+    
+    public init(repo: Repository, branch: Branch) {
```

Comment:
Ok, let's see if I understand you right. Because the code returns only Iterator of only "Commit" types object so it's too restrictive, while Commit, Tag, Branch are & can be equally viewed as the "Commit".

So, what's the function should return is of somewhat iterator of super class of those or some protocols those objects has in common, correct?